.github/workflows/build.yml

```diff
@@ -312,78 +345,412 @@ jobs:
-        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/rc'
+  macos-package-github:
+    name: MacOS Package GitHub Release Assets
+    runs-on: macos-latest
+    needs: macos-build
+    if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/rc' || github.ref == 'refs/heads/better-build-assets'
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+
+      - name: Set up Node
+        uses: actions/setup-node@46071b5c7a2e0c34e49c3cb8a0e792e86e18d5ea # v2.1.5
+        with:
+          node-version: '14'
+
+      - name: Cache Node Modules
+        id: node-modules-cache
+        uses: actions/cache@c64c572235d810460d0d6876e9c705ad5002b353 # v2.1.6
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-${{ github.run_id }}-node-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Set Node options
+        run: echo "NODE_OPTIONS=--max_old_space_size=4096" >> $GITHUB_ENV
+
+      - name: Update NPM
+        run: |
+          npm install -g npm@7
+          npm install -g node-gyp
+          node-gyp install $(node -v)
+
+      - name: Print environment
+        run: |
+          node --version
+          npm --version
+          echo "GitHub ref: $GITHUB_REF"
+          echo "GitHub event: $GITHUB_EVENT"
+
+      - name: Get Build Cache
+        id: build-cache
+        uses: actions/cache@c64c572235d810460d0d6876e9c705ad5002b353 # v2.1.6
+        with:
+          path: build
+          key: ${{ runner.os }}-${{ github.run_id }}-build
+
+      - name: Setup Safari Cache
+        id: safari-cache
+        uses: actions/cache@c64c572235d810460d0d6876e9c705ad5002b353 # v2.1.6
+        with:
+          path: dist-safari
+          key: ${{ runner.os }}-${{ github.run_id }}-safari-extension
+
+      - name: Decrypt secrets
+        run: ./.github/scripts/macos/decrypt-secrets.ps1
         shell: pwsh
-        run: ./scripts/safari-build.ps1 -copyonly
+        env:
+          DECRYPT_FILE_PASSWORD: ${{ secrets.DECRYPT_FILE_PASSWORD }}
 
-      - name: Build application (dev)
-        if: github.ref != 'refs/heads/master' || github.ref != 'refs/heads/rc'
+      - name: Set up keychain
+        run: ./.github/scripts/macos/setup-keychain.ps1
+        shell: pwsh
+        env:
+          KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+          DESKTOP_KEY_PASSWORD: ${{ secrets.DESKTOP_KEY_PASSWORD }}
+          DEVID_CERT_PASSWORD: ${{ secrets.DEVID_CERT_PASSWORD }}
+          APPSTORE_CERT_PASSWORD: ${{ secrets.APPSTORE_CERT_PASSWORD }}
+          MACDEV_CERT_PASSWORD: ${{ secrets.MACDEV_CERT_PASSWORD }}
+          APPLE_ID_PASSWORD: ${{ secrets.APPLE_ID_PASSWORD }}
+
+      - name: Set up provisioning profiles
+        run: ./.github/scripts/macos/setup-profiles.ps1
+        shell: pwsh
+
+      - name: Increment version
+        run: ./.github/scripts/macos/increment-version.ps1
+        shell: pwsh
+
```

Comment:
These are essentially setting environment variables to use later. Thoughts on doing this in the build step and setting them as outputs and then grabbing them in these jobs?

We can always do this when we come back around to do more cleanup